mlflow/store/_unity_catalog/registry/rest_store.py

```diff
@@ -682,8 +682,11 @@ def _local_model_dir(self, source, local_model_path):
                 # Clean up temporary model directory at end of block. We assume a temporary
                 # model directory was created if the `source` is not a local path
                 # (must be downloaded from remote to a temporary directory) and
-                # `local_model_dir` is not a UC volumes path
-                if not os.path.exists(source) and not _is_uc_volumes_path(local_model_dir):
+                # `local_model_dir` is not a FUSE-mounted path. The check for FUSE-mounted
+                # paths is important as mlflow.artifacts.download_artifacts() can return
+                # a FUSE mounted path equivalent to the (remote) source path in some cases,
+                # e.g. return /dbfs/some/path for source dbfs:/some/path.
+                if not os.path.exists(source) and not is_fuse_or_uc_volumes_uri(local_model_dir):
```

Comment:
@smurching Minor question, the original `_is_uc_volumns_path` also covers `"/volume:/.."` (starting from lower case), while the new one only validates capital case. Is it ok?
https://github.com/mlflow/mlflow/blob/430a2bf78fc31564bd20b9dae0f87036a49e6a0e/mlflow/utils/uri.py#L109